hokusai/cli/review_app.py

```diff
@@ -0,0 +1,61 @@
+from collections import OrderedDict
+import click
+import yaml
+
+import hokusai
+
+from hokusai.cli.base import base
+from hokusai.cli.staging import KUBE_CONTEXT
+from hokusai.lib.common import set_verbosity, CONTEXT_SETTINGS
+from hokusai.lib.config import config
+
+@base.group()
+def review_app(context_settings=CONTEXT_SETTINGS):
+  """Manage/Create review apps"""
+  pass
+
+@review_app.command(context_settings=CONTEXT_SETTINGS)
+@click.argument('app_name', type=click.STRING)
+@click.option('-v', '--verbose', type=click.BOOL, is_flag=True, help='Verbose output')
+@click.option('-sf', '--source-file', type=click.STRING, default='hokusai/staging.yml', help='Source deployment file')
+@click.option('-dn', '--destination-namespace', type=click.STRING, default=None, help='Target Namespace')
+def create(app_name, verbose, source_file, destination_namespace):
+  set_verbosity(verbose)
+  # copy staging yml file
+  if destination_namespace is None: destination_namespace = app_name
+  create_new_app_yaml(source_file, app_name, destination_namespace)
+
+@review_app.command(context_settings=CONTEXT_SETTINGS)
+@click.argument('app_name', type=click.STRING)
+@click.option('-v', '--verbose', type=click.BOOL, is_flag=True, help='Verbose output')
+def apply(app_name, verbose):
+  """Create the Kubernetes resources defined in ./hokusai/%s.yml""" % app_name
+  set_verbosity(verbose)
+  hokusai.k8s_create(KUBE_CONTEXT, app_name, app_name)
+
+
+def create_new_app_yaml(source_file, app_name, destination_namespace):
+  with open(source_file, 'r') as stream:
+    try:
+      yaml_content = list(yaml.load_all(stream))
+      # update namespace to destination namespace
+      for c in yaml_content: update_namespace(c, destination_namespace)
+      # prepend new namespace definition to yaml
+      new_namespace = OrderedDict([
+          ('apiVersion', 'v1'),
+          ('kind', 'Namespace'),
+          ('metadata', {
+            'name': str(app_name)
+          })
+        ])
+      yaml_content = [new_namespace] + yaml_content
+      with open("hokusai/%s.yml" % app_name, 'w') as output:
+        yaml.safe_dump_all(yaml_content, output, default_flow_style=False)
+    except yaml.YAMLError as exc:
+      print(exc)
+
+def update_namespace(yaml_section, destination_namespace):
+  if 'namespace' in yaml_section: yaml_section['namespace'] = destination_namespace
```

Comment:
Would just overwrite namespace here - i.e. drop the `if`

Reply:
but that way this will set `namespace` in each yaml section, we want to only update if that yaml portion has `namespace`, i kind of use this method recursively.

Reply:
Reading more [here](https://github.com/kubernetes/kubernetes/issues/7789) about what happens when a given namespace is defined in the `kubectl` context verses a given yaml spec.  This might be working for us in tests because we are using the `default` namespace in the `context` definition in our `~/.kube/config` file and operations against `yaml` resource specs without defining the namespace work as intended.  In any case it seems this is the behavior implemented:

> If no namespace is specified in the schema, then I'd expect the one from kubeconfig to be used.

> If a namespace is explicitly specified on the command line and that conflicts with the one in the schema, kubectl should raise an error.

This is why I am in favor of putting `namespace` in the schema as it takes precedence over implicit schemas provided by `context`